core-library/src/main/java/org/silverpeas/core/io/media/image/ImageTool.java

```diff
@@ -23,23 +23,36 @@
  */
 package org.silverpeas.core.io.media.image;
 
+import org.silverpeas.core.io.media.image.option.AbstractImageToolOption;
+import org.silverpeas.core.util.ServiceProvider;
+
 import java.io.File;
 import java.util.Set;
 
-import org.silverpeas.core.io.media.image.option.AbstractImageToolOption;
-
 /**
  * @author Yohann Chastagnier
  */
 public interface ImageTool {
 
+  static ImageTool get() {
+    return ServiceProvider.getService(ImageTool.class);
+  }
+
   /**
    * Indicates if image tools are available
    * @return
    */
   boolean isActivated();
 
   /**
+   * Gets aimed information from the given image file.
+   * @param source the image file.
+   * @param infoTypes the aimed types of info.
+   * @return
+   */
+  String[] getImageInfo(File source, ImageInfoType... infoTypes) throws Exception;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS00112)